architecture_docs/ADRs/0001-webhook_architecture.md

```diff
@@ -0,0 +1,66 @@
+
+Currently all Pelorus exporters like failure, committime and deploytime are designed and implemented to use `pull` model to gather the data from external sources such as GitHub, GitLab, Jira, ServiceNow, etc...
+
+The proposal is to allow Pelorus to receive data from the external data sources using, which will use `push` method to send data to the Pelorus.
+
+The benefits of the `push` method:
+ - The architecture to integrate 3rd party CI/CD systems is going to be simplified
+ - Standardized communication API between 3rd party systems and Pelorus
+ - Pelorus gaining the option to receive additional data such as SLA or SLO
+ - Possible reduction of the exporter instances, due to generic nature of the webhook exporter in which one instance may serve multiple purposes
+ - Less network traffic. Only when an event interesting from the Pelorus perspective happens the information will be sent
+ - Reduced number of the OCP priviledges required by the `push` method exporter. The `push` method does not require access to the various OCP resources for querying them
+
+## Decision
+
+We will use an HTTP webhook to implement Pelorus `push` medthod for receiving information.
+
+### Design Overview
+
+The webhook is an additional exporter, that exposes two services, one for receiving data from various external systems such as Jira, GitHub, GitLab, Bitbucket, and others. It then via the second service exposes this data to Prometheus, which will scrape and store the data for Pelorus purposes.
+
+To ensure seamless integration with 3rd party external systems and enforce data validation for the Project, the webhook exporter will include plugins that are specifically designed to communicate with external systems. These plugins will be tailored to the unique requirements of the Pelorus, allowing for efficient and accurate data transmission between the external systems and the Pelorus's Prometheus instance.
+
+### How it works
+
+1. Webhook is deployed as additional exporter on it's own or side to other Pelorus exporters
+2. External systems send data to the webhook via a webhook endpoint
+3. The webhook receives the data and transforms it into a Pelorus compatible format
+4. The webhook exposes the transformed data to Prometheus by exposing a Prometheus endpoint
+5. Prometheus scrapes the webhook's Prometheus endpoint and stores the received data in its time-series database
+
+```
+                +-----------------------+            +-----------------------+
+                |  External System (1)  |            |  External System (N)  |
+                +-----------+-----------+            +-----------+-----------+
+                            | (send)                             | (send)
+            +---------------+              +---------------------+----------+
+            |                              |                                |
++-----------v-----------+      +-----------v-----------+        +-----------v-----------+
```

Comment:
I like the image, but I thought you can use one instance of webhook to receive data from countless external systems. The image passes me the message you need one for each external system, like our exporters today. 

So, what is the intended way to use the webhook?

Reply:
The (N) in the "External System (N)" represents the "countless external systems".

The use case to have multiple webhooks is still valid for the scenario where the webhook is using secrets to configure the "Secret". For example here is [GitHub secret doc](https://docs.github.com/en/webhooks-and-events/webhooks/securing-your-webhooks#setting-your-secret-token).

For the scenario with "secret" if user would like to create many endpoints with different secrets then the user would need to create multiple instances of webhook to apply such secret to it.

Reply:
we can address this concern of mine in the webhook documentation